### ISSUE TITLE
[help] Training loss becoming nan?

### ISSUE THREAD

**Issue by kevin3567** [TUDB-Labs/MixLoRA#21]
I attempted to finetune a Llama-7B model using the MixLoRA code. However, About 1 epoch into the training, I started getting nan. What might be the cause?

Note that all training is conducted on a single 24GB GPU.

![Screenshot from 2024-12-02 16-40-51](https://github.com/user-attachments/assets/bb696982-3256-47f2-b490-680400dd1e62)

**Comment by mikecovlee** [comment]
Consider using a lower learning rate or checking your runtime environment (such as the driver). We have noticed that on some devices that do not support `bfloat16`, it will fall back to `float16`, which can easily cause gradient explosion and result in `NaN` values.

**Comment by kevin3567** [comment]
Fort it working now after a creating a fresh conda env and reinstalling all the requirements.
